### PR TITLE
decorate a class

### PR DESCRIPTION
**Pull request by christianbiring1**
In this project, I have used the Decorator design pattern to correct the names of people.

### CODE REVIEW

File: nameable.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
+  end
+end
+
+class Decorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    @nameable = nameable
+    super()
```
Comment (ShoiraTa) [inline nameable.rb:12]: - [ ] OPTIONAL: You do not need to call super here 😉

**Review by ShoiraTa — APPROVED** [review]
### **STATUS: APPROVED  ✔️** 
------
Hi @christianbiring1 ,  

Your project is complete! Great job for nailing it right on the first attempt 🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

giphy

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread by mentioning my Github username if something is not 100% clear.
***
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have  limited number of  reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: nameable.rb (hunk 1 above)
Comment (christianbiring1) [reply nameable.rb:12]: Thanks for the review @ShoiraTa !.. :rocket: 
but when I wasn't calling the `super` I was getting the linters errors that's why I have put it.

Comment (ShoiraTa) [reply nameable.rb:12]: Oh that's true, I understood, thanks 👍🏻

**Pull request merged**